tests/phpunit/CRM/Case/XMLProcessor/ProcessTest.php

```diff
@@ -0,0 +1,169 @@
+    $this->assertActivityAssignedToContactExists(NULL);
+  }
+
+  /**
+   * Tests the creation of activities with default assignee set to a specific contact.
+   */
+  public function testCreateActivityAssignedToSpecificContact() {
+    $this->activityTypeXml->default_assignee_type = $this->defaultAssigneeOptionsIds['SPECIFIC_CONTACT'];
+    $this->activityTypeXml->default_assignee_contact = $this->assigneeContactId;
+
+    $this->process->createActivity($this->activityTypeXml, $this->params);
+    $this->assertActivityAssignedToContactExists($this->assigneeContactId);
+  }
+
+  /**
+   * Tests the creation of activities with default assignee set to a specific contact,
+   * but the contact does not exist.
+   */
+  public function testCreateActivityAssignedToNonExistantSpecificContact() {
+    $this->activityTypeXml->default_assignee_type = $this->defaultAssigneeOptionsIds['SPECIFIC_CONTACT'];
+    $this->activityTypeXml->default_assignee_contact = 987456321;
+
+    $this->process->createActivity($this->activityTypeXml, $this->params);
+    $this->assertActivityAssignedToContactExists(NULL);
+  }
+
+  /**
+   * Tests the creation of activities with the default assignee being the one
+   * creating the case's activity.
+   */
+  public function testCreateActivityAssignedToUserCreatingTheCase() {
+    $this->activityTypeXml->default_assignee_type = $this->defaultAssigneeOptionsIds['USER_CREATING_THE_CASE'];
+
+    $this->process->createActivity($this->activityTypeXml, $this->params);
+    $this->assertActivityAssignedToContactExists($this->_loggedInUser);
+  }
+
+  /**
+   * Tests the creation of activities when the default assignee is set to NONE.
+   */
+  public function testCreateActivityAssignedNoUser() {
+    $this->activityTypeXml->default_assignee_type = $this->defaultAssigneeOptionsIds['NONE'];
+
+    $this->process->createActivity($this->activityTypeXml, $this->params);
+    $this->assertActivityAssignedToContactExists(NULL);
+  }
+
+  /**
+   * Tests the creation of activities when the default assignee is set to NONE.
+   */
+  public function testCreateActivityWithNoDefaultAssigneeOption() {
+    $this->process->createActivity($this->activityTypeXml, $this->params);
+    $this->assertActivityAssignedToContactExists(NULL);
+  }
+
+  /**
+   * Asserts that a an activity was created where the assignee was the one related
+   * to the target contact.
+   * It also deletes this activity from the test database.
+   *
+   * @param int|null the ID of the expected assigned contact or NULL if expected to be empty.
+   */
+  protected function assertActivityAssignedToContactExists($assigneeContactId) {
+    $activity = $this->callAPISuccess('Activity', 'getsingle', array(
+      'target_contact_id' => $this->targetContactId,
+      'assignee_contact_id' => $assigneeContactId
+    ));
+
+    $this->assertNotNull($activity, 'Contact has no activities assigned to them');
+
+    CRM_Activity_BAO_Activity::deleteActivityContact($activity['id'], 1);
```

Comment:
Although I see the advantage of having this here to avoid a second API call, it's not usual to have an assertion writing/deleting data. Imagine you want to assert this but want to also assert something else related to the activity?

For data cleanup, it's better to use the `tearDown()` method. Or, maybe, you can use the `Civi\Test\TransactionalInterface`, which resets any data created in a test automatically. We use it for extension tests, but I'm not sure if it works for core tests

Reply:
Actually this test extends from `CiviCaseTestCase` and that one has a `tearDown` function that truncastes the activity table so I'll just delete that one line and let the tearDown do its work.